pymdownx/magiclink.py

```diff
@@ -1036,6 +1045,33 @@ def handleMatch(self, m, data):
         return el, m.start(0), m.end(0)
 
 
+class MagiclinkCustomRefPattern(InlineProcessor):
+    """Return a link Element given a custom prefix."""
+
+    ANCESTOR_EXCLUDES = ('a',)
+
+    def __init__(self, pattern, md, shortname, target_url):
+        """Initialize."""
+
+        self.shortname = shortname
```

Comment:
TODO: better term than "shortname". This is the lower-cased version of the prefix that we inject into a class name and inject into the name we use when registering the inlineProcessor.